### ISSUE TITLE
Sponsors & Exhibitor page

### ISSUE THREAD

**Issue by hao-fang** [acl-org/acl-2020-virtual-conference#25]
**Number of Volunteers**: 3

TODO (Updated after #41)
- [x] Edit [sponsors.json](https://github.com/acl-org/acl-2020-virtual-conference-sitedata/blob/master/sponsors.json). The information below need to be provided by the sponsor. For demo purpose, just put some generic values.
  - [x] change the logos (needs to upload images to the folder `static/images/acl2020`
  - [x] add a `channel` field to record the RocketChat channel, currently set them to `example_sponsor` 
  - [x] add a `description` field to record the sponsor descriptions 
  - [x] add a `website` field to record the link to the external Sponsor's website
  - [x] add a `zoom_schedule` field (list) to record Zoom Room links and schedules for live QA sessions.
  - [x] add a `contact` field (list) to record contact person information.
  - [x] other, e.g., downloadable links
- [x] Edit `main.py` [here](https://github.com/acl-org/acl-2020-virtual-conference/blob/ee76ec9dfa96812defbf60910aea8129f3db3f72/main.py#L36) to make sure the `sponsor` data is properly loaded. Note `site_data` stores the raw `sponsors.json`, whereas `by_uid["sponsors"]` record a uid-to-sponsor lookup dict.
- [x] Edit [sponsors.html](https://github.com/acl-org/acl-2020-virtual-conference/blob/master/templates/sponsors.html) and `components.html` [here](https://github.com/acl-org/acl-2020-virtual-conference/blob/ee76ec9dfa96812defbf60910aea8129f3db3f72/templates/components.html#L228) to make sure the "Sponsors" page look nice. See [ICASSP 2020](https://2020.ieeeicassp-virtual.org/patron-exhibitor-virtual-space) for reference.
- [x] Edit [sponsor.html](https://github.com/acl-org/acl-2020-virtual-conference/blob/master/templates/sponsor.html) to make sure the individual sponsor pages look nice.
  - [x] Should show the sponsor level.
  - [x] Should show the sponsor logo.
  - [x] It seems that the page is auto-focused on the RocketChat channel. Should disable this behavior. 
    * It seems to work fine now??? Need more testing

**Comment by hao-fang** [comment]
@Chandrahasd Thanks for working on this. This is currently one of our top priority thing because sponsors need to see a working example to get some more ideas.
Please let me know if you have any questions.

@luanyi Can you send necessary information to @Chandrahasd you get from sponsors?
For missing information, we can put a placeholder field for demo purpose.

**Comment by hao-fang** [comment]
> [ ] Edit [sponsors.html](https://github.com/acl-org/acl-2020-virtual-conference/blob/master/templates/sponsors.html) and `components.html` [here](https://github.com/acl-org/acl-2020-virtual-conference/blob/ee76ec9dfa96812defbf60910aea8129f3db3f72/templates/components.html#L228) to make sure the "Sponsors" page look nice. See [ICASSP 2020](https://2020.ieeeicassp-virtual.org/patron-exhibitor-virtual-space) for reference.

@Chandrahasd This might be a good starter project.

**Comment by Chandrahasd** [comment]
@hao-fang Sure! I'll start with this one.

**Comment by hao-fang** [comment]
UPDATE about RocketChat channels (cc @luanyi)

We will use public **Broadcast Channels**.  Broadcasted channels behave like read-only channels, with only users with the right permission being able to post there.
* Users without permission (the same one to post on read-only channels) inside this channel won't be able to see each other in the user list.
* Users without permission won't be able to react to messages.
* Every message contains a reply button that redirects the user to a direct message with the user that posted the message.

Source: https://docs.rocket.chat/guides/user-guides/channels#creating-channels.

I've changed the current #example_sponsor to be a broadcast channel. You can use a test account to see what it looks like. (Your admin account will be able to post messages.)

**Comment by jcklie** [comment]
(cc @hao-fang ) I see that @Chandrahasd has already an open PR on `sponsors.html` so I do hesitate to make it more beautiful by myself. Can we take the images from the [ACL sponsor page](https://acl2020.org/sponsors/list/) and use them or do we need to ask sponsors to give them to us again? I would have a look to fix the style of `sponsor.html`. The other next steps I see are either blocked (sponsors.html) or need input (sponsor.html needs real data).

**Comment by hao-fang** [comment]
> (cc @hao-fang ) I see that @Chandrahasd has already an open PR on `sponsors.html` so I do hesitate to make it more beautiful by myself. Can we take the images from the [ACL sponsor page](https://acl2020.org/sponsors/list/) and use them or do we need to ask sponsors to give them to us again? I would have a look to fix the style of `sponsor.html`. The other next steps I see are either blocked (sponsors.html) or need input (sponsor.html needs real data).

@jcklie Yeah, I feel the best way is probably to have you start working on the `sponsor.html` -- there are more work to be done in those pages and I don't have a good intuition about how to arrange those components nicely. But I'll let you three to discuss with Yi to see how to collaborate efficiently :)

**Comment by hao-fang** [comment]
@jcklie @Chandrahasd @nickatomlin @luanyi 
The current page looks nice!
I briefly tested the https://virtual.acl2020.org/sponsor_3.html.
Here are some points I found.
- [ ] Make sure all clickable links are working.
   - [ ] "Visit Website" is not working
- [ ] Add `target="_blank"` to all clickable links so it opens a new "external" pages.
- [ ] It seems that after the page load, it still auto-focus on the RocketChat channel. Can we fix this?

**Comment by jcklie** [comment]
For the sponsors.html:

- [ ] If you resize the page, then the images overlap before hitting the responsive breakpoint to be one below each other
- [ ] Some transparent logos get a grey background, I would make the background white for all

**Comment by raosudha89** [comment]
@jcklie  @hao-fang  The sponsors page looks really nice! Is there something more to be done on this issue for basic website? If not, then can we close this and move remaining enhancement tasks to a new issue?

**Comment by hao-fang** [comment]
I think the remaining tasks here is to fill in the missing data for some sponsors, e.g., https://virtual.acl2020.org/sponsor_17.html is not complete.
I'll let people who are working on this to decide whether to close or keep this issue open.
For now, I move this to "blocked" in the the project view https://github.com/acl-org/acl-2020-virtual-conference/projects/2

**Comment by luanyi** [comment]
We are still waiting for information from several sponsors. The biggest remaining task is to add Zoom link to the virtual booths. I think it's better to keep this issue open at least during the dry run. The sponsors will raise some issues, and we need to adjust the virtual booths & correct some information accordingly.

**Comment by Chandrahasd** [comment]
Hi @jcklie, @hao-fang,

Just one minor concern. In the current sponsors' data, I have merged the following two pieces of information into a single "Resources" section.

```
7. Download (they can provide up to 5 pdf files)
9. A pdf or a link to the papers & presentation pointers can be submitted after the conference schedule is released. 
```

They are shown in a single section on the sponsors' page.

![Screen Shot 2020-06-27 at 11 57 01 PM](https://user-images.githubusercontent.com/6125084/85929511-1fd6c300-b8d3-11ea-9988-45506ed1cf25.png)

Let me know if we need to change it.

Thank you.

**Comment by hao-fang** [comment]
@Chandrahasd okay with me :)

**Comment by goodmami** [comment]
Just comparing the sponsors page with what's currently at https://acl2020.org/sponsors/list/, and I see that ByteDance is not listed in the Gold sponsors here. Should it be added?

**Comment by luanyi** [comment]
@goodmami ByteDance is not in the sponsor list Priscilla provided us. I have never received any email from them either so I don't think they are one of our sponsors. I'll double check with Priscilla. In any case it's already too late to prepare a virtual booth for them. We can put a logo and add a link to their company.
